src/main/java/fworks/da/PostgresAccessManager.java

```diff
@@ -98,129 +150,526 @@ public boolean checkIfCourseExists(String courseId) {
+     * @param userId          the user ID of the enrolment being deleted
+     */
+    @Override
+    public void removeCourseEnrolmentQuery(
+            String courseId,
+            String userId) {
+
+        String query = "DELETE FROM ec.enrolments " +
+                "WHERE user_id='" + userId + "' AND " +
+                "course_id='" + courseId + "';";
+
+        try (Statement statement = db.createStatement()) {
+            statement.executeQuery(query);
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /** Queries postgres DB to get the unique test ID corresponding to the given
+     * solution ID.
+     *
+     * @param inputId the solution ID of the solution document being queried
+     * @return a string representing a unique test ID
+     */
+    @Override
+    public String getTestIdBySolutionIdQuery(String inputId){
+
+        String testId;
+        String query = "SELECT test_id FROM ec.solutions " +
+                "WHERE solution_id='" + inputId + "';";
+
+        try (Statement statement = db.createStatement()) {
+            ResultSet rs = statement.executeQuery(query);
+            rs.next();
+
+            testId = rs.getString("test_id");
+
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+        return testId;
+    }
+
+    /** Queries postgres DB to get the unique course ID corresponding to the
+     * given test ID.
+     *
+     * @param inputId the test ID of the test document being queried
+     * @return a string representing a unique course ID
+     */
+    @Override
+    public String getCourseIdByTestIdQuery(String inputId){
+
+        String courseId;
+        String query = "SELECT course_id FROM ec.tests " +
+                "WHERE test_id='" + inputId + "';";
+
+        try (Statement statement = db.createStatement()) {
+            ResultSet rs = statement.executeQuery(query);
+            rs.next();
+
+            courseId = rs.getString("course_id");
+
         } catch (SQLException e) {
             throw new RuntimeException(e);
         }
-        return solutionIdList;
+        return courseId;
     }
+
+    /** Queries postgres DB to save data for new message entity.
+     *
+     * @param messageId     the message ID of the message entity being saved
+     * @param solutionId    the solution ID of the message entity being saved
+     * @param userId        the user ID of the message entity being saved
+     * @param parentId      the parent ID of the message entity being saved
+     * @param messageBody   the message body of the message entity being saved
+     */
+    @Override
+    public void addMessageQuery(String messageId,
+                                String solutionId,
+                                String userId,
+                                String parentId,
+                                String messageBody) {
+        String query = "INSERT INTO ec.messages(message_id, " +
```

Comment:
Would it be better if we generated the messageId (along with other PK IDs) within PostgresAccessManager? If not we would have to implement the random ID generator method in all the classes which are going to call these "add to database" methods.